packages/flutter_tools/lib/src/android/device_android.dart

```diff
@@ -529,3 +511,38 @@ String getAdbPath() {
     return _defaultAdbPath;
   }
 }
+
+class _AdbLogReader extends DeviceLogReader {
+  _AdbLogReader(this.device);
+
+  final AndroidDevice device;
+
+  String get name => 'Android';
+
+  Future<int> logs({bool clear: false}) async {
+    if (!device.isConnected())
+      return 2;
+
+    if (clear)
+      device.clearLogs();
+
+    return await runCommandAndStreamOutput(device.adbCommandForDevice([
+      'logcat',
+      '-v',
+      'tag', // Only log the tag and the message
+      '-s',
+      'flutter:V',
+      'ActivityManager:W',
+      'System.err:W',
+      '*:F',
+    ]), prefix: '[Android] ');
+  }
+
+  int get hashCode => name.hashCode;
+
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    return other is _AdbLogReader;
```

Comment:
Yeah, an adb log reader pulls from adb logcat, so it's output will be identical with another instance. Will add more docs -